### PR TITLE
added possibility of running tests on selenium grid on old ff

### PR DESCRIPTION
**Pull request by joannajasnowska**
## Description
I provided possibility to run tests via selenium grid - at first step selenium grid uses currently supported browser - ff 38.6

## Motivation and Context
This is first step to allow to run AET tests on chrome.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/FirefoxWebDriverFactory.java [hunk 1]
```diff
@@ -137,8 +149,10 @@ private void setCommonCapabilities(DesiredCapabilities capabilities, FirefoxProf
     capabilities.setCapability("firefox_binary", new File(path).getAbsolutePath());
   }
 
-  private WebDriver getFirefoxDriver(DesiredCapabilities capabilities) {
-    WebDriver driver = new FirefoxDriver(capabilities);
+  private WebDriver getFirefoxDriver(DesiredCapabilities capabilities)
+      throws MalformedURLException {
+    WebDriver driver = StringUtils.isNotBlank(seleniumGridUrl) ? new RemoteWebDriver(
+        new URL(seleniumGridUrl), capabilities) : new FirefoxDriver(capabilities);
```
Comment (Skejven) [inline core/worker/src/main/java/com/cognifide/aet/worker/drivers/FirefoxWebDriverFactory.java]: Maybe let's rename `FirefoxDriver` to `LocalFirefoxDriver` ?

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/FirefoxWebDriverFactory.java [hunk 2]
```diff
@@ -68,6 +76,10 @@
   @Property(name = LOG_FILE_PATH, label = "Path to firefox error log", value = DEFAULT_FIREFOX_ERROR_LOG_FILE_PATH)
   private String logFilePath;
 
+  @Property(name = SELENIUM_GRID_URL, label = "Url to selenium grid hub", value = DEFAULT_SELENIUM_GRID_URL)
```
Comment (Skejven) [inline core/worker/src/main/java/com/cognifide/aet/worker/drivers/FirefoxWebDriverFactory.java]: Please update the documentation and explain there the logic that is behind switching from local ff driver to selenium grid one.
For now it looks that the logic is, if user defines `seleniumGridUrl` parameter value, Selenium Grid is used.
Is that fair, that this one is preferred over `path` parameter?
This logic should be documented.

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/FirefoxWebDriverFactory.java (hunk 1 above)
Comment (joannajasnowska) [reply core/worker/src/main/java/com/cognifide/aet/worker/drivers/FirefoxWebDriverFactory.java]: FirefoxDriver comes from selenium, we would need to create additional class that extends it.
I'm not sure if it's worth it

File: documentation/src/main/wiki/LinuxAndWindowsSetup.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ Please note that full list of required tools and its versions can be found in [S
 ##### Linux Setup
 1. Turn off Firewall. This may be achieved differently on various linux distribution, for example on CentOS `selinux` and `iptables` should be disabled.
 2. Install MongoDB in version 3.2.3
-3. Install JDK (1.7) from Oracle or OpenJDK
+3. Install JDK (1.8) from Oracle or OpenJDK
```
Comment (wiiitek) [inline documentation/src/main/wiki/LinuxAndWindowsSetup.md]: 👍

**Comment by wiiitek** [comment]
Please wait with merging until the [AET cookbook changes](https://github.com/Cognifide/aet-cookbook/pull/15) are released.

**Pull request merged**